src/test/java/de/tum/in/www1/artemis/GradeStepIntegrationTest.java

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /**
+     * Test get request for a single grade step by grade percentage when no grading scale exists
+     *
+     * @throws Exception
+     */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetGradeStepByPercentageForCourseNoGradingScaleExists() throws Exception {
+        request.get("/api/courses/" + course.getId() + "/grading-scale/match-grade-step?gradePercentage=70", HttpStatus.NOT_FOUND, GradeStep.class);
+    }
+
+    /**
+     * Test get request for a single grade step by grade percentage
+     *
+     * @throws Exception
+     */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetGradeStepByPercentageForCourse() throws Exception {
+        GradeStep gradeStep = new GradeStep();
+        gradeStep.setGradeName("Name");
+        gradeStep.setLowerBoundPercentage(60);
+        gradeStep.setUpperBoundPercentage(100);
+        gradeStep.setGradingScale(courseGradingScale);
+        gradeSteps = Set.of(gradeStep);
+        courseGradingScale.setGradeSteps(gradeSteps);
+        gradingScaleRepository.save(courseGradingScale);
+
+        GradeStep foundGradeStep = request.get("/api/courses/" + course.getId() + "/grading-scale/match-grade-step?gradePercentage=70", HttpStatus.OK, GradeStep.class);
+
+        assertThat(foundGradeStep).usingRecursiveComparison().ignoringFields("gradingScale", "id").isEqualTo(gradeStep);
+    }
+
+    /**
+     * Test get request for a single grade step by grade percentage when no grading scale exists
+     *
+     * @throws Exception
+     */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetGradeStepByPercentageForExamNoGradingScaleExists() throws Exception {
+        request.get("/api/courses/" + course.getId() + "/exams/" + exam.getId() + "/grading-scale/match-grade-step?gradePercentage=70", HttpStatus.NOT_FOUND, GradeStep.class);
+    }
+
+    /**
+     * Test get request for a single grade step by grade percentage
+     *
+     * @throws Exception
+     */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetGradeStepByPercentageForExam() throws Exception {
+        GradeStep gradeStep = new GradeStep();
+        gradeStep.setGradeName("Name");
+        gradeStep.setLowerBoundPercentage(60);
+        gradeStep.setUpperBoundPercentage(100);
+        gradeStep.setGradingScale(examGradingScale);
+        gradeSteps = Set.of(gradeStep);
+        examGradingScale.setGradeSteps(gradeSteps);
+        gradingScaleRepository.save(examGradingScale);
+
+        GradeStep foundGradeStep = request.get("/api/courses/" + course.getId() + "/exams/" + exam.getId() + "/grading-scale/match-grade-step?gradePercentage=70", HttpStatus.OK,
+                GradeStep.class);
+
+        assertThat(foundGradeStep).usingRecursiveComparison().ignoringFields("gradingScale", "id").isEqualTo(gradeStep);
```

Comment:
Thats pretty cool :) Did not know about that recursive comparison!

Reply:
Yeah, it's pretty neat :)